src/main/java/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class SupplyOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public SupplyOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterSupply = fruitDao.getFruitBalance(fruit) + quantity;
+        if (balanceAfterSupply >= 0) {
+            fruitDao.updateBalanceOfFruit(fruit,balanceAfterSupply);
+        } else {
+            throw new RuntimeException("Quantity must be greater than zero."
+                    + " But you provide: " + quantity);
```

Comment:
The error message 'Quantity must be greater than zero.' is misleading in the context of a supply operation. Since the supply operation is supposed to add stock, the quantity should be allowed to be zero or greater. Please correct the error message or remove the unnecessary check.